lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -623,9 +624,22 @@ private static IActiveContext CreateMockActiveContext(int id)
         /// <returns></returns>
         private static IFailedTask CreateMockFailedTask(string taskId, string errorMsg)
         {
+            return CreateMockFailedTask(taskId, errorMsg, new Exception());
+        }
+
+        /// <summary>
+        /// Creates a mock FailedTask with specified taskId and error message
+        /// </summary>
+        /// <param name="taskId"></param>
+        /// <param name="errorMsg"></param>
+        /// <param name="taskException"></param>
+        /// <returns></returns>
+        private static IFailedTask CreateMockFailedTask(string taskId, string errorMsg, Exception taskException)
```

Comment:
Actually, it might be better to keep signature `CreateMockFailedTask(string taskId, string errorMsg)` and construct necessary exception class inside mock based on error message. Right now error message and exception type can be inconsistent, for example, in test `TestFailedNoRunningTaskBeforeAllTasksAreRunningScenario` error message used is `TaskGroupCommunicationError` but no exception parameter is passed, so when classifying this based on exception type it would record `FailedTaskSystemError`, not `FailedTaskCommunicationError`.